tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -29,26 +35,471 @@ def setUpClass(cls) -> None:
+                        'execution_date': self.now.isoformat(),
+                        'external_trigger': True,
+                        'start_date': self.now.isoformat(),
+                        'conf': {},
+                    },
+                    {
+                        'dag_id': 'TEST_DAG_ID',
+                        'dag_run_id': 'TEST_DAG_RUN_ID_2',
+                        'end_date': None,
+                        'state': 'running',
+                        'execution_date': (self.now + timedelta(minutes=1)).isoformat(),
+                        'external_trigger': True,
+                        'start_date': self.now.isoformat(),
+                        'conf': {},
+                    }
+                ],
+                "total_entries": 2
+            }
+        )
+
+    @provide_session
+    def test_handle_limit_in_query(self, session):
+        dagrun_models = [DagRun(
+            dag_id='TEST_DAG_ID',
+            run_id='TEST_DAG_RUN_ID' + str(i),
+            run_type=DagRunType.MANUAL.value,
+            execution_date=self.now + timedelta(minutes=i),
+            start_date=self.now,
+            external_trigger=True,
+        ) for i in range(100)]
+
+        session.add_all(dagrun_models)
+        session.commit()
+        result = session.query(DagRun).all()
+        assert len(result) == 100
+
+        response = self.client.get(
+            "api/v1/dags/TEST_DAG_ID/dagRuns?limit=1"
+        )
+        assert response.status_code == 200
+        self.assertEqual(response.json.get('total_entries'), 1)
+        self.assertEqual(
+            response.json,
+            {
+                "dag_runs": [
+                    {
+                        'dag_id': 'TEST_DAG_ID',
+                        'dag_run_id': 'TEST_DAG_RUN_ID0',
+                        'end_date': None,
+                        'state': 'running',
+                        'execution_date': (self.now + timedelta(minutes=0)).isoformat(),
+                        'external_trigger': True,
+                        'start_date': self.now.isoformat(),
+                        'conf': {},
+                    }
+                ],
+                "total_entries": 1
+            }
+        )
+
+    @provide_session
+    def test_handle_offset_in_query(self, session):
+        dagrun_models = [DagRun(
+            dag_id='TEST_DAG_ID',
+            run_id='TEST_DAG_RUN_ID' + str(i),
+            run_type=DagRunType.MANUAL.value,
+            execution_date=self.now + timedelta(minutes=i),
+            start_date=self.now,
+            external_trigger=True,
+        ) for i in range(4)]
+
+        session.add_all(dagrun_models)
+        session.commit()
+        result = session.query(DagRun).all()
+        assert len(result) == 4
+
+        response = self.client.get(
+            "api/v1/dags/TEST_DAG_ID/dagRuns?offset=1"
+        )
+        assert response.status_code == 200
+        self.assertEqual(response.json.get('total_entries'), 3)
```

Comment:
```suggestion
        self.assertEqual(response.json.get('total_entries'), 4)
```
This should always return the amount of all items.
<img width="497" alt="Screenshot 2020-06-08 at 11 44 19" src="https://user-images.githubusercontent.com/12058428/84016326-69d82300-a97d-11ea-952c-37c472d3e16f.png">

Reply:
Looks like the offset=1 is what is making it to return 3 instead of 4. I have tried different ways for the schema and total_entries but it always come as 3 instead of 4. I used pdb debugger and found out the offset is the cause. The query it runs is this:
    
    result2 = session.query(DagRun).offset(1).limit(100).all()
    assert len(result2) == 3

The number of DagRun in db is 4 and it offsets from 1 returning the rest.
Should I run a separate query that returns all entries and assign it to total_entries?

Reply:
Yes. Separate query sound good.

Reply:
Look what I found when I ran webserver
**Without Filter Top**
![withoutfiltertop](https://user-images.githubusercontent.com/4122866/84039791-af0a4e00-a999-11ea-9e14-dc183ae48d33.png)
**With Filter Top**
![withfiltertop](https://user-images.githubusercontent.com/4122866/84039923-db25cf00-a999-11ea-922b-87f76770338a.png)
**Without Filter Bottom**
![withoutfilterbottom](https://user-images.githubusercontent.com/4122866/84039990-f1338f80-a999-11ea-9fb1-5d4fdddc3a77.png)
**With Filter Bottom**
![withfilterbottom](https://user-images.githubusercontent.com/4122866/84040041-06102300-a99a-11ea-9df0-8b3cc861df71.png)

Reply:
I am thinking that we have been handling the total_entries correctly

Reply:
I will really appreciate if you could take a look at this once more

Reply:
Fixed [5096389](https://github.com/apache/airflow/pull/9153/commits/5096389c1447d8100e70fdab7b593bfd4112aaf9)